### ISSUE TITLE
ExpressionSet2phyloseq throwing warning

### ISSUE THREAD

**Issue by damagedcoda** [vobencha/curatedMetagenomicData#32]
Hey,

I want to work with the liver cirrhosis data set from the current version of your package. When I try to convert the ExpressionSet-Object into a phyloseq-Object, I get the following warning:

```
Warning message:
Too few values at 983 locations: 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, ... 
```

Here is my code:
```R
library(devtools)
# Use the devbranch version from github
install_github("vobencha/curatedMetagenomicData")
library(curatedMetagenomicData)

QinObject <- QinN_2014.metaphlan_bugs_list.stool()
QinObject <- ExpressionSet2phyloseq(QinObject)
```

The warning is pretty cryptic and I would like to know whats going on here. 
Thanks a lot,
Nicolai

Update 1:

I believe I have found something which is connected to this warning. Converting the above ExpressionSet-Object to a phyloseq Object also messes up the relative count data one obtains from MetaPhlAn. Observe the output of the following lines of code:

```R Abundance table from expressionSet Object
expressionSet_object <- QinN_2014.metaphlan_bugs_list.stool()
phyloseq_object <- ExpressionSet2phyloseq(QinObject)
head(exprs(expressionSet_object))
```
Output:
```R
                                  HD-1     HD-10     HD-11     HD-12     HD-13     HD-14     HD-15     HD-16     HD-17    HD-18     HD-19
k__Bacteria                   99.97458 100.00000 100.00000 100.00000 100.00000 100.00000 100.00000 100.00000 100.00000 99.57479 100.00000
k__Viruses                     0.02542   0.00000   0.00000   0.00000   0.00000   0.00000   0.00000   0.00000   0.00000  0.03003   0.00000
k__Bacteria|p__Bacteroidetes  73.10375  68.20391  41.61057  59.47165  65.24694  81.47807  62.88468  40.87083  56.19501 32.38622  50.73240
k__Bacteria|p__Firmicutes     24.44844  24.47002  47.48853  36.90053  33.23794  15.53996  28.24266  57.71649  34.52424 42.11154  43.93038
```
Note that this is what one would expect from a MetaPhlAn abundance profile: A hierarchical listing of relative abundances (Scaled to 100)

In contrast, the phyloseq_object:
```R
head(otu_table(phyloseq_object))
```
Output:
```R
OTU Table:          [6 taxa and 237 samples]
                     taxa are rows
                    HD-1   HD-10   HD-11   HD-12   HD-13   HD-14   HD-15   HD-16   HD-17  HD-18   HD-19    HD-2   HD-20   HD-21   HD-22   HD-23
k__Bacteria       999746 1000000 1000000 1000000 1000000 1000000 1000000 1000000 1000000 995748 1000000 1000000 1000000 1000000 1000000 1000000
k__Viruses           254       0       0       0       0       0       0       0       0    300       0       0       0       0       0       0
p__Bacteroidetes  731038  682039  416106  594716  652469  814781  628847  408708  561950 323862  507324  559771  130263  510156  312239  696036
p__Firmicutes     244484  244700  474885  369005  332379  155400  282427  577165  345242 421115  439304  386967  747054  343789  675458  269415
```

Observe that ExpressionSet2phyloseq converts the dezimal into an integer in a weird way. 

Cheers,
Nic

**Comment by lwaldron** [comment]
This is some odd thing that was introduced in @schifferl's re-factoring of my (perfectly functional :dancing_men:) `metaphlanToPhyloseq()` function, but which has no effect on the result:

```
> QinObject <- QinN_2014.metaphlan_bugs_list.stool()
snapshotDate(): 2016-12-06
see ?curatedMetagenomicData and browseVignettes('curatedMetagenomicData') for documentation
loading from cache ‘/Users/lw391//.ExperimentHub/325’
> o1=ExpressionSet2phyloseq(QinObject, rounding = FALSE) #warning
Warning message:
Too few values at 983 locations: 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, ... 
> source("https://raw.githubusercontent.com/waldronlab/presentations/master/Waldron_2016-06-07_EPIC/metaphlanToPhyloseq.R")
> o2=metaphlanToPhyloseq(tax=exprs(QinObject), metadat=pData(QinObject)) #no warning
> identical(o1, o2)
[1] TRUE
> 
```

It may relate to `ExpressionSet2phyloseq` re-using `phyloseq` function names as object names, but in any case we'll fix it because it's a disconcerting warning. Also I just noticed that the default for the "rounding" argument got changed from FALSE to TRUE, so users may be surprised to find all their relative abundances multiplied by 1e6. This the rounding argument should be removed, and replaced with a relab (default TRUE), which if FALSE multiplies by read depths to approximate counts.

**Comment by lwaldron** [comment]
Fixed in Pull Request #34

**Comment by lwaldron** [comment]
BTW in case you're curious, the culprit was `tidyr::separate()`, fixed by changing:

``        
separate(., ".", taxonomic.ranks, sep = "\\|") %>%
``

to:

``
separate(., ".", taxonomic.ranks, sep = "\\|", fill="right") %>%
``

which produces exactly the same output without the warning about filling in NAs in the taxonomy table to the right of the taxonomic level of each row.
